UpcomingMovies/TMDbConfig.swift

```diff
@@ -80,39 +72,25 @@ class TMDbConfig: NSObject, NSCoding {
     }
   }
   
-  // MARK: NSCoding
+  // MARK: Codable
   
-  private enum CodingKeys: String {
-    case baseImageUrl
-    case secureBaseImageUrl
-    case posterSizes
+  private enum CodingKeys: String, CodingKey {
+    case baseImageUrl = "base_url"
+    case secureBaseImageUrl = "secure_base_url"
+    case posterSizes = "poster_sizes"
     case dateUpdated
   }
   
-  required init(coder aDecoder: NSCoder) {
-    baseImageURLString = aDecoder.decodeObject(forKey: CodingKeys.baseImageUrl.rawValue) as! String
-    secureBaseImageURLString = aDecoder.decodeObject(forKey: CodingKeys.secureBaseImageUrl.rawValue) as! String
-    posterSizes = aDecoder.decodeObject(forKey: CodingKeys.posterSizes.rawValue) as! [String]
-    dateUpdated = aDecoder.decodeObject(forKey: CodingKeys.dateUpdated.rawValue) as? Date
-  }
-  
-  func encode(with aCoder: NSCoder) {
-    aCoder.encode(baseImageURLString, forKey: CodingKeys.baseImageUrl.rawValue)
-    aCoder.encode(secureBaseImageURLString, forKey: CodingKeys.secureBaseImageUrl.rawValue)
-    aCoder.encode(posterSizes, forKey: CodingKeys.posterSizes.rawValue)
-    aCoder.encode(dateUpdated, forKey: CodingKeys.dateUpdated.rawValue)
-  }
-  
   func save() {
-    NSKeyedArchiver.archiveRootObject(self, toFile: _fileURL.path)
+    let encoder = JSONEncoder()
+    if let json = try? encoder.encode(self) {
+        UserDefaults.standard.set(json, forKey: "TheMovieDB-Configuration")
+    }
   }
   
   class func unarchivedInstance() -> TMDbConfig? {
-    if FileManager.default.fileExists(atPath: _fileURL.path) {
-      return NSKeyedUnarchiver.unarchiveObject(withFile: _fileURL.path) as? TMDbConfig
-    } else {
-      return nil
-    }
+    let decoder = JSONDecoder()
+    return try? decoder.decode(self, from: UserDefaults.standard.data(forKey: "TheMovieDB-Configuration") ?? "".data(using: .utf8)!)
```

Comment:
Please do not use explicitly unwrapped optionals (`!`).

Reply:
`"".data(using: .utf8)` will always be non-optional.

Reply:
I mean something like this:
```swift
let decoder = JSONDecoder()
guard let data = UserDefaults.standard.data(forKey: tmdbConfigKey) else {
  return nil
}

return try? decoder.decode(self, from: data)
```